.github/workflows/build.yaml

```diff
@@ -1,29 +1,45 @@
 name: build
-on:
-  push:
-    branches:
-      - main
-  pull_request:
+
+# Running CI on push for all branches and PRs
+on: ["push", "pull_request"]
```

Comment:
Do we want to run it on push on every branch? this will potentially use up more CI minutes than we need to

Reply:
Was not 100% sure about this one, but I kinda like the idea of having an immediate feedback every time I push the code in my feature branch.

Was thinking the following:
- Generally the pipeline does not take too long to run and now we have a timeout of max 5 minutes
- In case it will start to cost us a lot, we could move to self-hosted runners.

WDYT ?

Reply:
I think we can achieve this with Open PR no? We should also have concurrency groups to cancel previous runs or else we will just have 2 runs for every push to a new PR

Reply:
We could yes, but then I would always need to open a PR even for a work in progress task, which in the end will achieve the same as running the pipeline every time I push code.
I could switch to a draft PR but at that point I don't remember if the pipeline runs or not. I seem to remember the pipeline runs only when the PR is moved from Draft to Ready.
Good point about the concurrency - going to add it